bitnami/contour/templates/service_helpers.yaml

```
@@ -131,6 +131,15 @@ The default envoy service spec
       {{- else if eq .Values.envoy.service.type "ClusterIP" }}
       nodePort: null
       {{- end }}
+    - name: metrics
+      port: {{ .Values.envoy.service.ports.metrics }}
+      protocol: TCP
+      targetPort: {{ .Values.envoy.service.targetPorts.metrics }}
+      {{- if and (or (eq .Values.envoy.service.type "NodePort") (eq .Values.envoy.service.type "LoadBalancer")) (not (empty .Values.envoy.service.nodePorts.metrics)) }}
+      nodePort: {{ .Values.envoy.service.nodePorts.https }}
+      {{- else if eq .Values.envoy.service.type "ClusterIP" }}
+      nodePort: null
+      {{- end }}
```

In order to not modify the default behavior and give users control over whether to expose or not the metrics port in their service, I think we should make this conditional to a value, for example `envoy.service.exposeMetrics` which defaults to `false`.

```suggestion
    {{- if .Values.envoy.service.exposeMetrics }}
    - name: metrics
      port: {{ .Values.envoy.service.ports.metrics }}
      protocol: TCP
      targetPort: {{ .Values.envoy.service.targetPorts.metrics }}
      {{- if and (or (eq .Values.envoy.service.type "NodePort") (eq .Values.envoy.service.type "LoadBalancer")) (not (empty .Values.envoy.service.nodePorts.metrics)) }}
      nodePort: {{ .Values.envoy.service.nodePorts.https }}
      {{- else if eq .Values.envoy.service.type "ClusterIP" }}
      nodePort: null
      {{- end }}
    {{- end }}
```